lib/web_ui/lib/src/engine/dom_renderer.dart

```diff
@@ -630,6 +629,17 @@ $_glassPaneTagName * {
     EnginePlatformDispatcher.instance._updateLocales();
   }
 
+  // Creates a [HostNode] into a `root` [html.Element].
+  HostNode _createHostNode(html.Element root) {
+    // maybe if (js_util.hasProperty(root, 'attachShadow')) ?
```

Comment:
Yes, let's avoid using exceptions for non-exceptional control flow. hasProperty is the way to go, I think.